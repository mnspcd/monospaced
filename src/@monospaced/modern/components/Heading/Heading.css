.mdrn .Heading {
  display: block;
  font-size: 1em;
  margin: 0;
}

.mdrn .Heading--xxxl {
  font-family: var(--font-display);
  font-size: calc(1em * var(--type-scale-xxxl));
  font-variant: lining-nums;
  line-height: calc(var(--leading-scale-xxxl) / var(--type-scale-xxxl));
}

.mdrn .Heading--xxl,
.mdrn .Heading--xl,
.mdrn .Heading--l,
.mdrn .Heading--m,
.mdrn .Heading--s {
  font-family: var(--font-monospaced);
  font-variant: slashed-zero;
}

.mdrn .Heading--xxl {
  font-size: calc(1em * var(--type-scale-xxl));
  line-height: calc(var(--leading-scale-xxl) / var(--type-scale-xxl));
}

.mdrn .Heading--xl {
  font-size: calc(1em * var(--type-scale-xl));
  line-height: calc(var(--leading-scale-xl) / var(--type-scale-xl));
}

.mdrn .Heading--l {
  font-size: calc(1em * var(--type-scale-l));
  line-height: calc(var(--leading) / var(--type-scale-l));
}

.mdrn .Heading--m,
.mdrn .Heading--s {
  font-size: calc(1em * var(--type-scale-m));
  line-height: var(--leading);
}

@media (--larger-viewport) {
  .mdrn .Heading--s {
    font-size: calc(1em * var(--type-scale-s));
    line-height: calc(var(--leading) / var(--type-scale-s));
  }
}
