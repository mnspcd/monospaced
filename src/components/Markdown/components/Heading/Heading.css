.Heading {
  display: block;
  font-size: 1em;
  margin: 0;
}

.Heading--xxxl {
  font-family: var(--font-display);
  font-size: calc(1em * var(--type-scale-xxxl));
  font-variant: lining-nums;
  line-height: calc(var(--leading-scale-xxxl) / var(--type-scale-xxxl));
}

.Heading--xxl,
.Heading--xl,
.Heading--l,
.Heading--m,
.Heading--s {
  font-family: var(--font-monospaced);
  font-variant: slashed-zero;
}

.Heading--xxl {
  font-size: calc(1em * var(--type-scale-xxl));
  line-height: calc(var(--leading-scale-xxl) / var(--type-scale-xxl));
}

.Heading--xl {
  font-size: calc(1em * var(--type-scale-xl));
  line-height: calc(var(--leading-scale-xl) / var(--type-scale-xl));
}

.Heading--l {
  font-size: calc(1em * var(--type-scale-l));
  line-height: calc(var(--leading) / var(--type-scale-l));
}

.Heading--m,
.Heading--s {
  font-size: calc(1em * var(--type-scale-m));
  line-height: var(--leading);
}

@media (--larger-viewport) {
  .Heading--s {
    font-size: calc(1em * var(--type-scale-s));
    line-height: calc(var(--leading) / var(--type-scale-s));
  }
}
