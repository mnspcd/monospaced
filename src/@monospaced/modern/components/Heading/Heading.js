import classNames from "classnames";
import PropTypes from "prop-types";
import React from "react";

import { colorTokens } from "../../tokens/color";

import "./Heading.css";

/**
 * Use `Heading` to display heading text.
 */
const Heading = ({ children, color = null, level = null, size = null }) => {
  let Element = "span";

  if (Heading.levels.includes(level)) {
    Element = `h${level}`;
  }

  return (
    <Element
      className={classNames({ Heading: true, [`Heading--${size}`]: size })}
      style={{ color: color && colorTokens[color] }}
    >
      {children}
    </Element>
  );
};

const colors = ["color-brand-primary-dark"];
const levels = ["1", "2", "3", "4", "5", "6"];
const sizes = ["s", "m", "l", "xl", "xxl", "xxxl"];

Heading.colors = colors;
Heading.levels = levels;
Heading.sizes = sizes;

Heading.propTypes = {
  /**
   * Text to display in the Heading.
   */
  children: PropTypes.node.isRequired,

  /**
   * Heading color variants.
   */
  color: PropTypes.oneOf(colors),

  /**
   * HTML section heading level. If not provided, the Heading will render as
   * a span.
   */
  level: PropTypes.oneOf(levels),

  /**
   * Typographic size of the Heading.
   */
  size: PropTypes.oneOf(sizes),
};

Heading.H1 = ({ children }) => (
  <Heading level="1" size="xxxl">
    {children}
  </Heading>
);

Heading.H1.displayName = "Heading.H1";

Heading.H1.propTypes = { children: PropTypes.node.isRequired };

Heading.H2 = ({ children }) => (
  <Heading level="2" size="xxl">
    {children}
  </Heading>
);

Heading.H2.displayName = "Heading.H2";

Heading.H2.propTypes = { children: PropTypes.node.isRequired };

Heading.H3 = ({ children }) => (
  <Heading level="3" size="xl">
    {children}
  </Heading>
);

Heading.H3.displayName = "Heading.H3";

Heading.H3.propTypes = { children: PropTypes.node.isRequired };

Heading.H4 = ({ children }) => (
  <Heading level="4" size="l">
    {children}
  </Heading>
);

Heading.H4.displayName = "Heading.H4";

Heading.H4.propTypes = { children: PropTypes.node.isRequired };

Heading.H5 = ({ children }) => (
  <Heading level="5" size="m">
    {children}
  </Heading>
);

Heading.H5.displayName = "Heading.H5";

Heading.H5.propTypes = { children: PropTypes.node.isRequired };

Heading.H6 = ({ children }) => (
  <Heading level="6" size="s">
    {children}
  </Heading>
);

Heading.H6.displayName = "Heading.H6";

Heading.H6.propTypes = { children: PropTypes.node.isRequired };

export default Heading;
