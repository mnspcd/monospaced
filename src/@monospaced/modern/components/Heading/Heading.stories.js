import { select, text } from "@storybook/addon-knobs";
import React from "react";

import Heading from "./Heading";

export default {
  component: Heading,
  title: "Components/Heading",
};

export const main = () => (
  <>
    <Heading size="xxxl">Hamburgefonstiv</Heading>
    <Heading size="xxl">Hamburgefonstiv</Heading>
    <Heading size="xl">Hamburgefonstiv</Heading>
    <Heading size="l">Hamburgefonstiv</Heading>
    <Heading size="m">Hamburgefonstiv</Heading>
    <Heading size="s">Hamburgefonstiv</Heading>
  </>
);

export const h1 = () => <Heading.H1>Hamburgefonstiv</Heading.H1>;

export const h2 = () => <Heading.H2>Hamburgefonstiv</Heading.H2>;

export const h3 = () => <Heading.H3>Hamburgefonstiv</Heading.H3>;

export const h4 = () => <Heading.H4>Hamburgefonstiv</Heading.H4>;

export const h5 = () => <Heading.H5>Hamburgefonstiv</Heading.H5>;

export const h6 = () => <Heading.H6>Hamburgefonstiv</Heading.H6>;

export const level = () => (
  <Heading level="1" size="xl">
    Hamburgefonstiv
  </Heading>
);

export const color = () => (
  <Heading color="color-brand-primary-dark" size="xl">
    Hamburgefonstiv
  </Heading>
);

export const basic = () => <Heading>Hamburgefonstiv</Heading>;

basic.story = { parameters: { docs: { disable: true } } };

export const knobs = () => (
  <div>
    <Heading
      color={select("color", [""].concat(Heading.colors), "") || null}
      level={select("level", [""].concat(Heading.levels), "") || null}
      size={select("size", Heading.sizes, "xl")}
    >
      {text("text", "Hamburgefonstiv")}
    </Heading>
  </div>
);

knobs.story = { parameters: { docs: { disable: true } } };
