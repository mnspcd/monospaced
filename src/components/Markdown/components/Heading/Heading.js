import PropTypes from "prop-types";
import React from "react";
import classNames from "classnames";

import "./Heading.css";
import { colorTokens } from "../../../../system/color";

const Heading = ({ children, color, level, size }) => {
  let Element = "span";

  if (Heading.levels.includes(level)) {
    Element = `h${level}`;
  }

  return (
    <Element
      className={classNames({ Heading: true, [`Heading--${size}`]: size })}
      style={{ color: color && colorTokens[color] }}
    >
      {children}
    </Element>
  );
};

Heading.H1 = ({ children }) => (
  <Heading level="1" size="xxxl">
    {children}
  </Heading>
);

Heading.H2 = ({ children }) => (
  <Heading level="2" size="xxl">
    {children}
  </Heading>
);

Heading.H3 = ({ children }) => (
  <Heading level="3" size="xl">
    {children}
  </Heading>
);

Heading.H4 = ({ children }) => (
  <Heading level="4" size="l">
    {children}
  </Heading>
);

Heading.H5 = ({ children }) => (
  <Heading level="5" size="m">
    {children}
  </Heading>
);

Heading.H6 = ({ children }) => (
  <Heading level="6" size="s">
    {children}
  </Heading>
);

Heading.colors = ["brand-primary-dark-color"];
Heading.levels = ["1", "2", "3", "4", "5", "6"];
Heading.sizes = ["s", "m", "l", "xl", "xxl", "xxxl"];

Heading.propTypes = {
  children: PropTypes.node.isRequired,
  color: PropTypes.oneOf(Heading.colors),
  level: PropTypes.oneOf(Heading.levels),
  size: PropTypes.oneOf(Heading.sizes),
};

export default Heading;
