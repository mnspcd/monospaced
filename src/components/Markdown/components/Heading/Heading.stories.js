import React from "react";
import { select, text } from "@storybook/addon-knobs";
import { storiesOf } from "@storybook/react";

import Grid from "../../../Grid";
import Space from "../../../Space";

import Heading from "./Heading";

const defaultText = "Hamburgefonstiv";

storiesOf("Components/Markdown/Heading", module)
  .add("Component", () => (
    <div>
      <Heading
        color={select("color", [""].concat(Heading.colors), "") || null}
        level={select("level", [""].concat(Heading.levels), "") || null}
        size={select("size", Heading.sizes, "xl")}
      >
        {text("text", defaultText)}
      </Heading>
    </div>
  ))
  .add("Waterfall", () => (
    <Grid>
      <Grid.Item colSpan="6">
        <Space />
        {Heading.sizes
          .map(size => (
            <Heading key={size} size={size}>
              {text("text", defaultText)}
            </Heading>
          ))
          .reverse()}
      </Grid.Item>
    </Grid>
  ));
