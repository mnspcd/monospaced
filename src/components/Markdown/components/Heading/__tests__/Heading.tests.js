/* eslint-env jest */
import React from "react";
import { render } from "enzyme";

import Heading from "../Heading";

describe("Heading component", () => {
  it("should render correctly", () => {
    const component = render(<Heading>children</Heading>);
    expect(component).toMatchSnapshot();
  });

  it("should render heading color correctly", () => {
    const component = render(
      <Heading color="brand-primary-dark-color">children</Heading>,
    );
    expect(component).toMatchSnapshot();
  });

  it("should render heading level correctly", () => {
    const component = render(<Heading level="1">children</Heading>);
    expect(component).toMatchSnapshot();
  });

  it("should render an H1 correctly", () => {
    const component = render(<Heading.H1>children</Heading.H1>);
    expect(component).toMatchSnapshot();
  });

  it("should render an H2 correctly", () => {
    const component = render(<Heading.H2>children</Heading.H2>);
    expect(component).toMatchSnapshot();
  });

  it("should render an H3 correctly", () => {
    const component = render(<Heading.H3>children</Heading.H3>);
    expect(component).toMatchSnapshot();
  });

  it("should render an H4 correctly", () => {
    const component = render(<Heading.H4>children</Heading.H4>);
    expect(component).toMatchSnapshot();
  });

  it("should render an H5 correctly", () => {
    const component = render(<Heading.H5>children</Heading.H5>);
    expect(component).toMatchSnapshot();
  });

  it("should render an H6 correctly", () => {
    const component = render(<Heading.H6>children</Heading.H6>);
    expect(component).toMatchSnapshot();
  });
});
