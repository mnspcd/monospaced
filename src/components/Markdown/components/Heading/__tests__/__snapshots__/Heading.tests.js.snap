// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading component should render an H1 correctly 1`] = `
<h1
  class="Heading Heading--xxxl"
>
  children
</h1>
`;

exports[`Heading component should render an H2 correctly 1`] = `
<h2
  class="Heading Heading--xxl"
>
  children
</h2>
`;

exports[`Heading component should render an H3 correctly 1`] = `
<h3
  class="Heading Heading--xl"
>
  children
</h3>
`;

exports[`Heading component should render an H4 correctly 1`] = `
<h4
  class="Heading Heading--l"
>
  children
</h4>
`;

exports[`Heading component should render an H5 correctly 1`] = `
<h5
  class="Heading Heading--m"
>
  children
</h5>
`;

exports[`Heading component should render an H6 correctly 1`] = `
<h6
  class="Heading Heading--s"
>
  children
</h6>
`;

exports[`Heading component should render correctly 1`] = `
<span
  class="Heading"
>
  children
</span>
`;

exports[`Heading component should render heading color correctly 1`] = `
<span
  class="Heading"
  style="color:#000a6e"
>
  children
</span>
`;

exports[`Heading component should render heading level correctly 1`] = `
<h1
  class="Heading"
>
  children
</h1>
`;
